docs/custom_quantum_functions.md

```diff
@@ -156,11 +172,47 @@ void matrix_init_user(void) {
 }
 ```
 
+### `keyboard_pre_init_*` Function Documentation
+
+* Keyboard/Revision: `void keyboard_pre_init_kb(void)`
+* Keymap: `void keyboard_pre_init_user(void)`
+
+## Matrix Initialization Code
+
+This is called when the matrix is initialized, and after some of the hardware has been set up, but before many of the features have been initialized. 
+
+This is useful for setting up stuff that you may need elsewhere, but isn't hardware related nor is dependant on where it's started. 
+
+
 ### `matrix_init_*` Function Documentation
 
 * Keyboard/Revision: `void matrix_init_kb(void)`
 * Keymap: `void matrix_init_user(void)`
 
+
+## Keyboard Post Initialization code
```

Comment:
I'm not sure I like the word "Post" here. That implies that initialization should be complete by this point. I don't have any immediate suggestions for better names, but to me the naming of this means that I have no idea what the purpose of this function is. It's post initialization, so nothing should be initialized here, but it's also not called in a loop or anything?

Reply:
Perhaps `user_code_init`?

Reply:
The biggest issue with the name is that there are 3 inits.  Which ... is far from ideal. 

Though, the post init is the last function run as part of the keyboard setup, and should always be the last thing called. 

The problem with `user_code_init` is... the functionality levels. `user_code_init_kb`, and `user_code_init_user`? 

init_cleanup? init_finished? 

I'm not great with naming, and this was close in name to existing stuff.

Reply:
Well, one question: does there need to be a `_kb` and `_user` version of this function? What would the keyboard need to do that it doesn't already have a callback for?